error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/IsInstanceUsage.java

```diff
@@ -0,0 +1,57 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.LinkType.CUSTOM;
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+import static com.google.errorprone.BugPattern.StandardTags.STYLE;
+import static com.sun.source.tree.Tree.Kind.INSTANCE_OF;
+import static com.sun.source.tree.Tree.Kind.LAMBDA_EXPRESSION;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.LambdaExpressionTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.LambdaExpressionTree;
+
+/** A {@link BugChecker} that aligns usages of T.class::isInstance. */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary = "Use Class::isInstance where possible",
+    link = BUG_PATTERNS_BASE_URL + "IsInstanceUsage",
+    linkType = CUSTOM,
+    severity = SUGGESTION,
+    tags = STYLE)
+public final class IsInstanceUsage extends BugChecker implements LambdaExpressionTreeMatcher {
+  private static final long serialVersionUID = 1L;
+
+  /** Instantiates a new {@link IsInstanceUsage} instance. */
+  public IsInstanceUsage() {}
+
+  @Override
+  public Description matchLambdaExpression(LambdaExpressionTree tree, VisitorState state) {
+    if (LAMBDA_EXPRESSION == tree.getKind() && INSTANCE_OF == tree.getBody().getKind()) {
+      return constructDescription(
+          tree, state, constructFix(tree, ((InstanceOfTree) tree.getBody()).getType()));
+    }
+    return Description.NO_MATCH;
+  }
+
+  private Description constructDescription(
+      LambdaExpressionTree tree, VisitorState state, SuggestedFix.Builder fixBuilder) {
+    SuggestedFix fix = fixBuilder.build();
+    return SuggestedFixes.compilesWithFix(
+            fix, state, ImmutableList.of(), /* onlyInSameCompilationUnit= */ true)
```

Comment:
Aye, `compilesWithFix` is very convenient, but also _very_ expensive. Unfortunately we'll have to find a way to avoid this. @Ptijohn which issue did you hit without this?

(In case you're wondering: indeed the `MethodReferenceUsage` check also relies on `compilesWithFix`, and this is why it hasn't been enabled yet for the internal code base. It's the primary reason for the work on [this branch](https://github.com/PicnicSupermarket/error-prone-support/compare/sschroevers/MethodReferenceUsageCheck-improvements).)

Reply:
To be honest? It's mainly that I'm discovering the tool 😅 (and thus I didn't know about the limitations of this one)
I will dig to see what's best in this case 👌

Reply:
Oh so IIUC you just exact that we can just drop it right? 
Will push it :) 
I indeed thought for a sec that the `compilexWithFix` was a mandatory step in the validation of such a mechanism, but makes sense that it's not the case was "simpler" changes like this one.